java-spiffe-core/src/test/java/io/spiffe/svid/jwtsvid/JwtSvidParseAndValidateTest.java

```diff
@@ -119,119 +131,159 @@ void testParseAndValidate_nullAudience_throwsNullPointerException() throws JwtSv
                         .expectedJwtSvid(newJwtSvidInstance(
                                 trustDomain.newSpiffeId("host"),
                                 audience,
                                 expiration,
-                                claims.getClaims(), TestUtils.generateToken(claims, key1, "authority1") ))
+                                claims.getClaims(), TestUtils.generateToken(claims, key1, "authority1", "JOSE") ))
                         .build()),
                 Arguments.of(TestCase.builder()
-                        .name("2. success using RSA signature")
+                        .name("2. using RSA signature")
                         .jwtBundle(jwtBundle)
                         .expectedAudience(audience)
-                        .generateToken(() -> TestUtils.generateToken(claims, key3, "authority3"))
+                        .generateToken(() -> TestUtils.generateToken(claims, key3, "authority3", "JWT"))
                         .expectedException(null)
                         .expectedJwtSvid(newJwtSvidInstance(
                                 trustDomain.newSpiffeId("host"),
                                 audience,
                                 expiration,
-                                claims.getClaims(), TestUtils.generateToken(claims, key3, "authority3")))
+                                claims.getClaims(), TestUtils.generateToken(claims, key3, "authority3", "JWT")))
                         .build()),
                 Arguments.of(TestCase.builder()
-                        .name("3. malformed")
+                        .name("3. using empty typ")
+                        .jwtBundle(jwtBundle)
+                        .expectedAudience(audience)
+                        .generateToken(() -> TestUtils.generateToken(claims, key3, "authority3", ""))
+                        .expectedException(null)
+                        .expectedJwtSvid(newJwtSvidInstance(
+                                trustDomain.newSpiffeId("host"),
+                                audience,
+                                expiration,
+                                claims.getClaims(), TestUtils.generateToken(claims, key3, "authority3")))
+                        .build())
+        );
+    }
+
+    static Stream<Arguments> provideFailureScenarios() {
+        KeyPair key1 = TestUtils.generateECKeyPair(Curve.P_521);
+        KeyPair key2 = TestUtils.generateECKeyPair(Curve.P_521);
+        KeyPair key3 = TestUtils.generateRSAKeyPair(2048);
+
+        TrustDomain trustDomain = TrustDomain.of("test.domain");
+        JwtBundle jwtBundle = new JwtBundle(trustDomain);
+        jwtBundle.putJwtAuthority("authority1", key1.getPublic());
+        jwtBundle.putJwtAuthority("authority2", key2.getPublic());
+        jwtBundle.putJwtAuthority("authority3", key3.getPublic());
+
+        SpiffeId spiffeId = trustDomain.newSpiffeId("host");
+        Date expiration = new Date(System.currentTimeMillis() + 3600000);
```

Comment:
nit: express the constant time in the future as a multiplication so it's a little easier to understand the value
```suggestion
        Date expiration = new Date(System.currentTimeMillis() + (60 * 60 * 1000));
```